notebooks/Medical Image Generation with Generative AI.ipynb

```diff
@@ -227,11 +227,22 @@
     "    \"action\": \"generate\",\n",
     "    \"specs\": {\n",
     "        \"num_output_samples\": 1,                    # Number of output samples\n",
-    "        \"body_region\": [\"chest\"],                   # Body region (“please refer to the list above for the supported nody regions)\n",
-    "        \"organ_list\": [\"liver\"],                    # Organs (please refer to the list above for the supported organs)\n",
-    "        \"num_inference_steps\": num_inference_steps  # Number of inference steps (usually don't need to set)\n",
+    "        \"body_region\": [\"chest\"],                   # Body region (“please refer to the list above for the supported body regions)\n",
+    "        \"anatomy_list\": [\"liver\"],                  # Anatomy to be segmented (please refer to the list above for the supported anatomies)\n",
     "    },\n",
     "}\n",
+    "\n",
+    "# for the integration test only\n",
```

Comment:
Hi @drbeh ! Please add this into the test script (test_monai_notebook.py) instead of having it in the notebook. Thanks!

Reply:
To achieve this, you may need `nbformat` to manipulate the cells. For example, this is what I do locally for development mode testing

```
def remove_download(cells, inplace=True): 
    # New cell content
    cell_copy = cells.copy() if not inplace else cells
    # Identify and replace the target cell
    def condition(cell):
        if cell.cell_type != "code":
            return False
        return cell.source.startswith("# Download the job artifacts, including logs, scripts, and model checkpoints")
    for cell in cell_copy:
        if condition(cell):
            cell.source = ""
            break  # Assuming there's only one cell with this marker, break after finding it
 
 nb = nbformat.read(notebook_path, as_version=4)
cells = nb.cells
cells = remove_download(cells)
```

Reply:
Hi @mingxin-zheng, I see your point but the way that integration tests are suppose to parametrize these notebooks are through the `parameters` cell (the cell tagged with "parameters") and the test cases are defined in `NOTEBOOK_TEST_CASES` as a list of dictionaries. Modifying arbitrary cells will break the logic and make it particular to this test.
It makes sense for what you have done which is not notebook dependents but adding specific content to a specific cell of a specific notebooks looks very hacky to me and it is hard to maintain in a long run. What to you think? @mingxin-zheng @Nic-Ma @yiheng-wang-nv

Reply:
@mingxin-zheng following your suggestion. I added a general rule to overwrite any tagged cell with the requested data. I will test to see if it works.

Reply:
I tested it and it worked for generative ai notebook.